lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -54,6 +54,7 @@ def purchase(money, payment_method, options = {})
         add_transaction(post, options)
         add_card(post, payment_method, options)
         add_card_present(post, options)
+        add_customer(post, options, payment_method)
```

Comment:
After changing the `add_customer` methods signature, you'll need to swap the `options` and `payment_method` params. Same for line 70 and 119.

Reply:
Thanks @naashton , I have updated this change 👍